python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -204,6 +204,12 @@ def _fsproxy_fast_to_slow(self):
             return fast_to_slow(self._fsproxy_wrapped)
         return self._fsproxy_wrapped
 
+    def as_gpu_object(self):
+        return self._fsproxy_slow_to_fast()
```

Comment:
Does it make sense to call `isinstance_cudf_pandas` inside this function and raise a friendly error message on False?